File: Proyecto en java/Mejor_Egresado_UTPL-Manuel Gomez/src/mejor_egresado_utpl/manuel/gomez/Mejor_Egresado_UTPLManuelGomez.java
package mejor_egresado_utpl.manuel.gomez;
import java.util.Scanner;
public class Mejor_Egresado_UTPLManuelGomez {
    public static void main(String[] args) {
        String nomEstudiante, estudianteEgresado = "", estudianteEgresado2 = "";
        int numEstudiantes, cont = 1, contCiclos, numCiclos;
        double promedioFinalEstudiante = 0, promedioEgresado = 0, promCiclo, promCiclos;
        boolean egresadosIguales = false;
        Scanner teclado = new Scanner(System.in);
        System.out.println("=================================== MEJOR EGRESADO UTPL ===================================");
        System.out.println("-Ingresar la Cantidad de Estudiantes");
        numEstudiantes = teclado.nextInt();
        while (cont <= numEstudiantes) {
            System.out.println("-Ingresar los NOMBRES y APELLIDOS del Estudiante");
            teclado.nextLine();
            nomEstudiante = teclado.nextLine();
            System.out.println("-Ingresar el Numero de Ciclos de la Carrera del Estudiante");
            numCiclos = teclado.nextInt();
            contCiclos = 1;
            promCiclos = 0;
            while (contCiclos <= numCiclos) {
                System.out.println("-Ingresar el Promedio Final del Ciclo "+contCiclos);
                promCiclo = teclado.nextDouble();
                promCiclos = (promCiclos + promCiclo);
                promedioFinalEstudiante = (promCiclos / numCiclos);
                contCiclos++;
            }
            if (promedioFinalEstudiante > promedioEgresado) {
                estudianteEgresado = nomEstudiante;
                promedioEgresado = promedioFinalEstudiante;
                egresadosIguales = false;
            }
            else if (promedioFinalEstudiante == promedioEgresado) {
                estudianteEgresado2 = nomEstudiante;
                egresadosIguales = true;
            }
            cont++;
        }
        System.out.println("=================================== MEJOR EGRESADO UTPL ===================================");
        if (egresadosIguales == true) {
            System.out.println("Los Mejores Egresados UTPL son: ");
            System.out.println("El estudiante "+estudianteEgresado+" por tener un Promedio Final de "+promedioEgresado);
            System.out.println("El estudiante "+estudianteEgresado2+" por tener un Promedio FInal de "+promedioEgresado);
        }
        else {
            System.out.println("El Mejor Egresado UTPL es: ");
            System.out.println("El estudiante "+estudianteEgresado+" por tener un Promedio Final de "+promedioEgresado);
        }
        System.out.println("===========================================================================================");
    }
}
//Desarrollado por Manuel Gomez

File: Proyecto en python/Mejor_Egresado_UTPL-Manuel Gomez.py
print("=================================== MEJOR EGRESADO UTPL ===================================")
print("-Ingresar la Cantidad de Estudiantes")
numEstudiantes = int(input())
promedioEgresado = 0
promedioFinalEstudiante = 0
egresadosIguales = False
cont = 1
while cont <= numEstudiantes:
    print("-Ingresar los NOMBRES y APELLIDOS del Estudiante")
    nomEstudiante = input()
    print("-Ingresar el Numero de Ciclos de la Carrera del Estudiante")
    numCiclos = int(input())
    contCiclos = 1
    promCiclos = 0
    while contCiclos <= numCiclos:
        print("-Ingresar el Promedio Final del Ciclo ",contCiclos)
        promCiclo = float(input())
        promCiclos = (promCiclos + promCiclo)
        promedioFinalEstudiante = (promCiclos / numCiclos)
        contCiclos +=1    
    if promedioFinalEstudiante > promedioEgresado:
        estudianteEgresado = nomEstudiante
        promedioEgresado = promedioFinalEstudiante
        egresadosIguales = False
    elif promedioFinalEstudiante == promedioEgresado:
        estudianteEgresado2 = nomEstudiante
        egresadosIguales = True
    cont +=1
print("=================================== MEJOR EGRESADO UTPL ===================================")
if egresadosIguales == True:
    print("Los Mejores Egresados UTPL son:")
    print("El estudiante",estudianteEgresado,"por tener un Promedio Final de",promedioEgresado)
    print("El estudiante",estudianteEgresado2,"por tener un Promedio Final de",promedioEgresado)
else:
    print("El Mejor Egresado UTPL es:")
    print("El estudiante",estudianteEgresado,"por tener un Promedio Final de",promedioEgresado)
print("===========================================================================================")